### PR TITLE
Allow watchdog probe timeout overrides

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR:

- adds possibility to configure watchdog probe timeouts https://github.com/giantswarm/giantswarm/issues/29983

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Test on Workload cluster.

### CODE REVIEW

File: helm/prometheus-agent/charts/prometheus-agent/files/watchdog.sh [hunk 1]
```diff
@@ -3,7 +3,9 @@
 ### Config
 
 metrics_url="localhost:9090/metrics"
-overload_tolerance="${WATCHDOG_OVERLOAD_TOLERANCE:-2}" # allow up to x times max shards
+overload_tolerance="${WATCHDOG_OVERLOAD_TOLERANCE:-5}" # allow up to x times max shards
+probe_timeout="${WATCHDOG_PROBE_TIMEOUT:-3}" # set the probe timeout to x seconds
+probe_retries=3 # set the number of retries for the liveness probe to x
```
Comment (hervenicol) [inline helm/prometheus-agent/charts/prometheus-agent/files/watchdog.sh:8]: Why not adding the possibility to override this one as well?

**Comment by QuentinBisson** [comment]
What do you mean by k8s would timeout ?

**Comment by hervenicol** [comment]
If the probe command does not return ever, there must be a time when kubernetes just gives up and says that the pod is broken, right?

Reading this doc, it seems that's 1 second. So I guess/hope we already override it somewhere!
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes

**Comment by QuentinBisson** [comment]
But then how come ours Can timeout after 2-3 seconds ?

**Comment by QuentinBisson** [comment]
It could bé yes, i'll édit thé pr on monday to use the computed values

**Review by hervenicol — APPROVED** [review]
LGTM

**Comment by QuentinBisson** [comment]
Now we have this:

```
    livenessProbe:
      exec:
        command:
        - /bin/sh
        - -c
        - WATCHDOG_OVERLOAD_TOLERANCE=5 WATCHDOG_PROBE_TIMEOUT=3 WATCHDOG_PROBE_RETRIES=3
          sh /etc/prometheus/configmaps/prometheus-agent-watchdog/watchdog.sh
      failureThreshold: 6
      periodSeconds: 5
      successThreshold: 1
      timeoutSeconds: 27
```

**Pull request merged**